cli/command/system/prune.go

```diff
@@ -0,0 +1,144 @@
+package system
+
+import (
+	"fmt"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/cli"
+	"github.com/docker/docker/cli/command"
+	units "github.com/docker/go-units"
+	"github.com/spf13/cobra"
+	"golang.org/x/net/context"
+)
+
+type pruneOptions struct {
+	force      bool
+	aggressive bool
+}
+
+// NewPruneCommand creates a new cobra.Command for `docker du`
+func NewPruneCommand(dockerCli *command.DockerCli) *cobra.Command {
+	var opts pruneOptions
+
+	cmd := &cobra.Command{
+		Use:   "prune [COMMAND]",
+		Short: "Delete unused data",
+		Args:  cli.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runPrune(dockerCli, true, true, true, opts)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.BoolVarP(&opts.force, "force", "f", false, "Do not prompt for confirmation")
+	flags.BoolVarP(&opts.aggressive, "aggressive", "a", false, "Delete all unused images not just dangling ones")
+
+	cmd.AddCommand(
+		&cobra.Command{
+			Use:   "containers",
+			Short: "Delete unused containers",
+			Args:  cli.NoArgs,
+			RunE: func(cmd *cobra.Command, args []string) error {
+				return runPrune(dockerCli, true, false, false, opts)
+			},
+		},
+		&cobra.Command{
+			Use:   "volumes",
+			Short: "Delete unused volumes",
+			Args:  cli.NoArgs,
+			RunE: func(cmd *cobra.Command, args []string) error {
+				return runPrune(dockerCli, false, true, false, opts)
+			},
+		},
+		&cobra.Command{
+			Use:   "images",
+			Short: "Delete unused images",
+			Args:  cli.NoArgs,
+			RunE: func(cmd *cobra.Command, args []string) error {
+				return runPrune(dockerCli, false, false, true, opts)
+			},
+		},
+	)
```

Comment:
So both `docker volume prune` and `docker system prune volumes` exist ?

Do we need both?  I think I prefer just having `docker x prune`. When we start to add filtering flags it's going to be a mess trying to support all the flags in this `system prune` command.